qiskit/circuit/library/standard_gates/x.py

```diff
@@ -1188,60 +1196,132 @@ def _define(self):
         definition = []
 
         if self._dirty_ancillas:
-            i = self.num_ctrl_qubits - 3
-            ancilla_pre_rule = [
-                (U2Gate(0, numpy.pi), [q_target], []),
-                (CXGate(), [q_target, q_ancillas[i]], []),
-                (U1Gate(-numpy.pi / 4), [q_ancillas[i]], []),
-                (CXGate(), [q_controls[-1], q_ancillas[i]], []),
-                (U1Gate(numpy.pi / 4), [q_ancillas[i]], []),
-                (CXGate(), [q_target, q_ancillas[i]], []),
-                (U1Gate(-numpy.pi / 4), [q_ancillas[i]], []),
-                (CXGate(), [q_controls[-1], q_ancillas[i]], []),
-                (U1Gate(numpy.pi / 4), [q_ancillas[i]], []),
-            ]
-            for inst in ancilla_pre_rule:
-                definition.append(inst)
-
-            for j in reversed(range(2, self.num_ctrl_qubits - 1)):
+            if self.num_ctrl_qubits < 3:
+                definition.append((MCXGate(self.num_ctrl_qubits), [*q_controls, q_target], []))
+            elif not self._relative_phase and self.num_ctrl_qubits == 3:
+                definition.append((C3XGate(), [*q_controls, q_target], []))
+            else:
+                num_ancillas = self.num_ctrl_qubits - 2
```

Comment:
To make this code a little easier to read and make it easier to potentially move the code to `qiskit.synthesis` at a later stage, could you put this synthesis technique into a private function (could be in this file) and then just call that function here? 🙂

Reply:
When calling the private function on the `qiskit.synthesis` we must create a new file in a new directory?

Reply:
In the meanwhile, you don't need to move the code to `qiskit.synthesis`. Just organize it in internal functions as we suggested above, so that later it will be easier for us to move it if needed.

Reply:
The idea is that not all this code will be inside the `define` function.
You can put the existing code inside `_synth_MCXVChain_basic` and the new code in `_synth_MCXVChain_isometry`, 
and let the `define` function call the new `_synth_MCXVChain_isometry`.